### ISSUE TITLE
SPContentDatabase triggers machine reboot via DSC

### ISSUE THREAD

**Issue by shurick81** [PowerShell/SharePointDsc#745]
**Details of the scenario you try and problem that is occurring:**
We are trying to upgrade content databases from SP 2013 to SP 2016. Each upgraded database require reboot.

**The DSC configuration that is using the resource:**
```
                    SPContentDatabase "ContentDB$dbName"
                    {
                        Name                    = $dbName
                        Ensure                  = "Present"
                        WebAppUrl               = "http://$webAppHostNameInsidan"
                        DatabaseServer          = $configParameters.SPDatabaseAlias
                        PsDscRunAsCredential    = $SPInstallAccountCredential
                        DependsOn               = "[SPWebApplication]InsidanWebApp"
                    }
```
**Version of the Operating System and PowerShell the DSC Target Node is running:**
Windows Server 2016, WMF 5.1
Azure environment

**Version of SharePoint that is used (e.g. SharePoint 2016):**
DBs were moved from SP 2013, SP1, CU 2017 May, English version + Swedish Language pack.
We're running script on SP 2016, CU 2017 October, English version + Swedish Language pack.

**Version of the DSC module you're using:**
1.9.0.0

**Comment by ykuijs** [comment]
There is no code in SPContentDatabase that forces this reboot. I suspect that this might be triggered by Mount-SPContentDatabasse.

Does this also happen if you manually run Mount-SPContentDatabase?

**Comment by shurick81** [comment]
Thank you for your reply @ykuijs. When we run `Mount-SPContentDatabase` it does not reboot the machine.

**Comment by shurick81** [comment]
Hi. I also checked with this script https://gallery.technet.microsoft.com/scriptcenter/Get-PendingReboot-Query-bdb79542 and it shows the following output after I mount via traditional PS:

```
Computer           : TST01SP16
CBServicing        : False
WindowsUpdate      : False
CCMClientSDK       :
PendComputerRename : False
PendFileRename     : False
PendFileRenVal     :
RebootPending      : False
```

**Comment by ykuijs** [comment]
@shurick81 Are you 100% sure the SPContentDatabase triggers the reboots? There is no code in SPContentDatabase that triggers a reboot, so I have no clue where this is coming from. Have you also tested with v2.0?

**Comment by shurick81** [comment]
Hello @ykuijs, I'm not really sure how to make sure 100% :).
What I see is this:
```
VERBOSE: [tst01sp16]: LCM:  [ Start  Resource ]  [[SPContentDatabase]ContentDBSP_Intra_Content_Portal]
VERBOSE: [tst01sp16]: LCM:  [ Start  Test     ]  [[SPContentDatabase]ContentDBSP_Intra_Content_Portal]
VERBOSE: [tst01sp16]:                            [[SPContentDatabase]ContentDBSP_Intra_Content_Portal] Testing content database configuration settings
VERBOSE: [tst01sp16]:                            [[SPContentDatabase]ContentDBSP_Intra_Content_Portal] Getting content database configuration settings
VERBOSE: [tst01sp16]:                            [[SPContentDatabase]ContentDBSP_Intra_Content_Portal] Executing as the local run as user UNIONEN\_spadm16
VERBOSE: [tst01sp16]:                            [[SPContentDatabase]ContentDBSP_Intra_Content_Portal] Leaving BeginProcessing Method of Get-SPDatabase.
VERBOSE: [tst01sp16]:                            [[SPContentDatabase]ContentDBSP_Intra_Content_Portal] Leaving ProcessRecord Method of Get-SPDatabase.
VERBOSE: [tst01sp16]:                            [[SPContentDatabase]ContentDBSP_Intra_Content_Portal] Leaving EndProcessing Method of Get-SPDatabase.
VERBOSE: [tst01sp16]: LCM:  [ End    Test     ]  [[SPContentDatabase]ContentDBSP_Intra_Content_Portal]  in 0.3750 seconds.
VERBOSE: [tst01sp16]: LCM:  [ Start  Set      ]  [[SPContentDatabase]ContentDBSP_Intra_Content_Portal]
VERBOSE: [tst01sp16]:                            [[SPContentDatabase]ContentDBSP_Intra_Content_Portal] Setting content database configuration settings
VERBOSE: [tst01sp16]:                            [[SPContentDatabase]ContentDBSP_Intra_Content_Portal] Executing as the local run as user UNIONEN\_spadm16
VERBOSE: [tst01sp16]:                            [[SPContentDatabase]ContentDBSP_Intra_Content_Portal] Leaving BeginProcessing Method of Get-SPDatabase.
VERBOSE: [tst01sp16]:                            [[SPContentDatabase]ContentDBSP_Intra_Content_Portal] Leaving ProcessRecord Method of Get-SPDatabase.
VERBOSE: [tst01sp16]:                            [[SPContentDatabase]ContentDBSP_Intra_Content_Portal] Leaving EndProcessing Method of Get-SPDatabase.
VERBOSE: [tst01sp16]:                            [[SPContentDatabase]ContentDBSP_Intra_Content_Portal] Leaving BeginProcessing Method of Get-SPWebApplication.
VERBOSE: [tst01sp16]:                            [[SPContentDatabase]ContentDBSP_Intra_Content_Portal] Leaving ProcessRecord Method of Get-SPWebApplication.
VERBOSE: [tst01sp16]:                            [[SPContentDatabase]ContentDBSP_Intra_Content_Portal] Leaving EndProcessing Method of Get-SPWebApplication.
VERBOSE: [tst01sp16]:                            [[SPContentDatabase]ContentDBSP_Intra_Content_Portal] Leaving BeginProcessing Method of Mount-SPContentDatabase.
VERBOSE: [tst01sp16]:                            [[SPContentDatabase]ContentDBSP_Intra_Content_Portal] Getting the web application
VERBOSE: [tst01sp16]:                            [[SPContentDatabase]ContentDBSP_Intra_Content_Portal] Found web application: Insidan
VERBOSE: [tst01sp16]:                            [[SPContentDatabase]ContentDBSP_Intra_Content_Portal] Getting the database server
VERBOSE: [tst01sp16]:                            [[SPContentDatabase]ContentDBSP_Intra_Content_Portal] Found database server address: SPDB
VERBOSE: [tst01sp16]:                            [[SPContentDatabase]ContentDBSP_Intra_Content_Portal] Performing the operation "Mount-SPContentDatabase" on target "SP_Intra_Content_Portal".
VERBOSE: [tst01sp16]:                            [[SPContentDatabase]ContentDBSP_Intra_Content_Portal] Mounting content database
WARNING: [tst01sp16]:                            [[SPContentDatabase]ContentDBSP_Intra_Content_Portal] Check C:\SPLogs\ULS for upgrade progress
VERBOSE: [tst01sp16]:                            [[SPContentDatabase]ContentDBSP_Intra_Content_Portal] New Content Database Id: d89e4303-461c-42de-9105-d5069d9b83f9
VERBOSE: [tst01sp16]:                            [[SPContentDatabase]ContentDBSP_Intra_Content_Portal] Leaving ProcessRecord Method of Mount-SPContentDatabase.
VERBOSE: [tst01sp16]:                            [[SPContentDatabase]ContentDBSP_Intra_Content_Portal] Leaving EndProcessing Method of Mount-SPContentDatabase.
VERBOSE: [tst01sp16]:                            [[SPContentDatabase]ContentDBSP_Intra_Content_Portal] Detected an update conflict, restarting server to allow DSC to resume and retry
VERBOSE: [tst01sp16]: LCM:  [ End    Set      ]  [[SPContentDatabase]ContentDBSP_Intra_Content_Portal]  in 131.0470 seconds.
VERBOSE: [tst01sp16]: LCM:  [ End    Resource ]  [[SPContentDatabase]ContentDBSP_Intra_Content_Portal]
VERBOSE: [tst01sp16]:                            [] A reboot is scheduled to progress further. Configuration will be continued after the reboot.
WARNING: [tst01sp16]:                            [] A reboot is scheduled to progress further. Configuration will be continued after the reboot.
VERBOSE: [tst01sp16]: LCM:  [ End    Set      ]
VERBOSE: [tst01sp16]: LCM:  [ End    Set      ]    in  147.2510 seconds.
VERBOSE: Operation 'Invoke CimMethod' complete.
VERBOSE: Time taken for configuration job to complete is 147.514 seconds
```

I tried out with 1.9 and now just checked that I have the same with 2.1

**Comment by ykuijs** [comment]
Ok, this really helped. I found where the reboot is coming from:
The message "Detected an update conflict, restarting server to allow DSC to resume and retry" can be found in the our helper module. This is generated when an update conflict is detected and to prevent issues the reboot is triggered.

After mounting the database, the code is trying to update the status of the database. I suspect that this throws the exception and therefore triggers this reboot. Do you specify the Enabled parameter in your config? If so can you remove that and try again?

**Comment by ykuijs** [comment]
Better yet, can you try to update the SPContentDatabase module as follows:

Update line 315:
```
                if ($params.Enabled -ne $cdbenabled)
```

with

```
                if ($params.ContainsKey("Enabled") -eq $true -and `
                    $params.Enabled -ne $cdbenabled)
```

on all servers and then try again?

**Comment by ykuijs** [comment]
@shurick81 Were you able to test the suggested update? If so I can add this to my next PR!

**Comment by shurick81** [comment]
Hi, not really, we implemented it via invoking Mount command in powershell script and I didn't check your proposal, sorry.

**Comment by ykuijs** [comment]
No worries, I just submitted the PR with this fix anyways.

**Comment by ykuijs** [comment]
Closed via PR #775
